rdmd/args.d

```diff
@@ -0,0 +1,29 @@
+/**
+Provides functionality for parsing command-line arguments and data
+structures for storing the results
```

Comment:
I should ask: are there any particular requirements for module documentation?  In particular, copyright and license notices?

Reply:
Thanks for bringing this up. I think they should at least have the license header, as in the old `rdmd.d`.

I think it would be fine to update the copyright notice to make DFL the author, as has been done with DMD and Phobos recently.

Reply:
DFL?

Reply:
DLF*, sorry. [D Language Foundation](https://dlang.org/foundation/).

Apparently I was wrong about Phobos, it was just DMD.
See e.g. https://github.com/dlang/dmd/blob/master/src/dmd/attrib.d#L5

Reply:
Given that there's no contributor agreement that I'm aware of, I'm kind of reluctant to attribute either authorship or copyright to an organization.  At least, _I_ don't want to be the one introducing that claim. ;-)  But I will try to write up some copyright/author headers.

Reply:
OK, feel free to reuse the existing header as is, (maybe add yourself to the authors).

Reply:
I was thinking of adding Jonathan Marler as the author of the `rdmd.filesystem` module, since that's his code.  I'll add myself only to modules where it makes sense (i.e. I'm making some meaningful contribution rather than just small edits and copy-pastes).

Reply:
> Given that there's no contributor agreement that I'm aware of, I'm kind of reluctant to attribute either authorship or copyright to an organization.

Well the way I see it is that at the moment the copyright is assigned to @andralex and I'm pretty sure that he is in favor of a reassignment to the DLF.

Reply:
AFAIK, any contributions to dlang repositories require giving copyright ownership to DFL...

Reply:
Since when?

Reply:
@wilzbach 
> Well the way I see it is that at the moment the copyright is assigned to @andralex and I'm pretty sure that he is in favor of a reassignment to the DLF.

rdmd's copyright notice says `Copyright (C) 2008 by Andrei Alexandrescu`.  But that's only because no one has bothered to maintain it since the code was first written.

Andrei can do with the copyright of his code however he wishes, but there have been many contributors over the years, some of them presumably significant, and without any official copyright transfer process, copyright for their contributions will remain theirs.

@marler8997 
> AFAIK, any contributions to dlang repositories require giving copyright ownership to DFL...

I am not aware of any such policy, and have never assigned any copyright over any of my contributions to dlang.  I believe that Walter may have had a copyright assignment requirement in place for the compiler backend, but there has never to my awareness been anything for the frontend, druntime, phobos, or the tools.  (After all, with everything Boost-licensed, what's the point of adding an extra bureaucratic step?)

This kind of stuff isn't done by implicit understanding.  Without explicit statements of copyright transfer, there is no basis for assuming copyright belongs solely to Andrei, or the DLF.

Reply:
https://github.com/dlang/dmd/blob/5b144ebf0fa4cecc21662eae9db1c1a608292e4d/README.md

> All significant contributors to DMD source code, via github, bugzilla, email, wiki, the D forums, etc., please assign copyright to those DMD source code changes to the D Language Foundation. Please send an email to walter@digitalmars.com with the statement:
>
>"I hereby assign copyright in my contributions to DMD to the D Language Foundation"
>
>and include your name and date.

This is what I remember reading from the DMD repo.  May not be the same policy for all repos.

Reply:
I wonder how many have sent an email like that.